monai/apps/reconstruction/networks/nets/utils.py

```diff
@@ -0,0 +1,182 @@
+        A tuple containing
+            (1) normalized output of shape (B,C,H,W) for 2D data or (B,C,H,W,D) for 3D data
+            (2) mean
+            (3) std
+    """
+    if len(x.shape) == 4:  # this is 2D
+        b, c, h, w = x.shape
+        x = x.contiguous().view(b, 2, c // 2 * h * w)
+        mean = x.mean(dim=2).view(b, 2, 1, 1, 1).expand(b, 2, c // 2, 1, 1).contiguous().view(b, c, 1, 1)
+        std = x.std(dim=2, unbiased=False).view(b, 2, 1, 1, 1).expand(b, 2, c // 2, 1, 1).contiguous().view(b, c, 1, 1)
+        x = x.view(b, c, h, w)
+        return (x - mean) / std, mean, std
+
+    elif len(x.shape) == 5:  # this is 3D
+        b, c, h, w, d = x.shape
+        x = x.contiguous().view(b, 2, c // 2 * h * w * d)
+        mean = x.mean(dim=2).view(b, 2, 1, 1, 1, 1).expand(b, 2, c // 2, 1, 1, 1).contiguous().view(b, c, 1, 1, 1)
+        std = x.std(dim=2, unbiased=False).view(b, 2, 1, 1, 1, 1).expand(b, 2, c // 2, 1, 1, 1).contiguous().view(b, c, 1, 1, 1)
+        x = x.view(b, c, h, w, d)
+        return (x - mean) / std, mean, std
+
+
+def reverse_complex_normalize(self, x: Tensor, mean: float, std: float) -> Tensor:
+    """
+    Reverses the normalization done by complex_normalize
+
+    Args:
+        x: input of shape (B,C,H,W) for 2D data or (B,C,H,W,D) for 3D data
+        mean: mean before normalization
+        std: std before normalization
+
+    Returns:
+        denormalized output of shape (B,C,H,W) for 2D data or (B,C,H,W,D) for 3D data
+    """
+    return x * std + mean
+
+
+def floor_ceil(n: float) -> Sequence:
+    """
+    Returns floor and ceil of the input
+
+    Args:
+        n: input number
+
+    Returns:
+        A tuple containing:
+            (1) floor(n)
+            (2) ceil(n)
+    """
+    return math.floor(n), math.ceil(n)
+
+
+def pad(self, x: Tensor) -> Sequence:
+    """
+    Pad input to feed into the network
+
+    Args:
+        x: input of shape (B,C,H,W) for 2D data or (B,C,H,W,D) for 3D data
+
+    Returns:
+        A tuple containing
+            (1) padded input
+            (2) pad sizes (in order to reverse padding if needed)
+    """
+    if len(x.shape) == 4:  # this is 2D
+        b, c, h, w = x.shape
+        w_mult = ((w - 1) | 15) + 1
+        h_mult = ((h - 1) | 15) + 1
+        w_pad = floor_ceil((w_mult - w) / 2)
+        h_pad = floor_ceil((h_mult - h) / 2)
+        x = F.pad(x, w_pad + h_pad)  # type: ignore
+        return x, (h_pad, w_pad, h_mult, w_mult)
+
+    elif len(x.shape) == 5:  # this is 3D
+        b, c, h, w, d = x.shape
+        w_mult = ((w - 1) | 15) + 1
+        h_mult = ((h - 1) | 15) + 1
+        d_mult = ((d - 1) | 15) + 1
+        w_pad = floor_ceil((w_mult - w) / 2)
+        h_pad = floor_ceil((h_mult - h) / 2)
+        d_pad = floor_ceil((d_mult - d) / 2)
+        x = F.pad(x, w_pad + h_pad + d_pad)
+        return x, (h_pad, w_pad, d_pad, h_mult, w_mult, d_mult)
+
+
+def reverse_pad(self, x: Tensor, pad_sizes: Sequence) -> Tensor:
+    """
```

Comment:
Could you help check whether `SpatialCrop()` may work here? https://github.com/Project-MONAI/MONAI/blob/46d9026c1ad5182ea8edc780771eb31929947b7c/monai/transforms/croppad/array.py#L465

Thanks

Reply:
Or may be the `inverse()` of `DivisiblePad` or `SpatialPad` might work?